packages/core/src/domain/tracekit/computeStackTrace.ts

```diff
@@ -104,9 +104,8 @@ const GECKO_EVAL_RE = /(\S+) line (\d+)(?: > eval line \d+)* > eval/i
 
 function parseGeckoLine(line: string): StackFrame | undefined {
   const parts = GECKO_LINE_RE.exec(line)
-  if (!parts) {
-    return
-  }
+  if (!parts) return
+  if (parts[3] === line && line !== '[native code]') return // overly greedy regex likely means this is a custom error message (exception native code)
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏ Ideally, the `parseGeckoLine` should not be used to parse React Native stacks :D anyway.

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏Can I suggest an alternative fix? Let's be pragmatical here. The issue seems to be that:
* the error message is included in the `error.stack` property
* and the error message is considered as a stack "line" by our regexes.

Because the error message is often crafted by the application, this can happen unpredictibly. The message could literally be `    at namedFunc0 (http://example.com/js/script.js:10:2)` and we wouldn't be able to distinghuish it from a normal stack line. What we could do instead is, before trying to parse the `error.stack` value, test if it contains the error message and remove it from the stack if that's the case. Something like this:

```ts
let stackProperty = tryToGetString(ex, 'stack')
const exString = String(ex)
if (startsWith(stack, exString) {
  exProperty = stack.slice(exString.length)
}
stackProperty.split('\n').forEach(...)
```

WDYT?

Reply:
`parseGeckoLine` is currently catching a lot of cases unrelated to Gecko parsing errors (`'[native code]'` is just one of the examples). This is due to a greedy regex used (the main culprit is `.*?` just after `\[native)` in the `GECKO_LINE_RE`). 

Renaming or breaking down this part of the regex would be ideal to better understand what is getting matched.

Reply:
This is a great suggestion, I tried out something similar previously (but with different assumptions). 

The issue is that in some cases, the first line and the stack trace are the same, and this solution will inadvertently remove the string.

A lot of this argument is based on unit tests that failed. I suspect that despite it's flaws, `parseGeckoLine` does not regex match these edge cases.

Reply:
> The issue is that in some cases, the first line and the stack trace are the same, and this solution will inadvertently remove the string.

Which case? Do you have something in mind? I suspect that it would not happen very often, right? It would be really particular, hand crafted errors and stacks. And even if it happens, we would still send all the data (message + stack): even if the stack would miss its first line, it would be in the message anyway, so I think this is acceptable.

Reply:
🤦‍♂️ `stack.slice` should have been `stackProperty.slice`. The solution you suggest actually solves this edge case without breaking anything else. Will update and add tests 

I still think we should maybe revisit `parseGeckoLine` at some point